src/v3/src/types/widget.ts

```diff
@@ -107,7 +110,10 @@ export type PageTitleCallback = (context: EventContext, param: PageTitleCallback
 export type OktaWidgetEventType = 'ready' | 'afterError' | 'afterRender';
 export type IDPDisplayType = 'PRIMARY' | 'SECONDARY';
 
-export type WidgetProps = Partial<WidgetOptions>;
+export type WidgetProps = Partial<WidgetOptions> & {
+  eventEmitter: EventEmitter;
+  widgetHooks: WidgetHooks; // instance of class WidgetHooks
```

Comment:
As I understand, `WidgetOptions` are "public" options that can be passed to `OktaSignIn` constructor. 
And `WidgetProps` are props passed to `<Widget />`. 
`eventEmitter` and `widgetHooks` are not public, so I moved them to `WidgetProps` (and marked as required as [you suggested](https://github.com/okta/okta-signin-widget/pull/3376#discussion_r1338846208)).

But maybe we can just use
```ts
export type WidgetProps = WidgetOptions;
```
since `Partial` is not needed
and move `eventEmitter` and `widgetHooks`  to `WidgetOptions` as optional. For simplicity.
@lesterchoi-okta ?

Reply:
Oh you're right, it might be right to keep `WidgetOptions` and `WidgetProps` separate the way you already have them

Reply:
I wrote a utility type yesterday that might be helpful:
```ts
type PartialExcept<T, K extends keyof T> = Partial<Omit<T, K>> & Pick<T, K>;
```